pkg/scheduler/core/assignment.go

```diff
@@ -48,14 +48,28 @@ func assignByAggregatedStrategy(state *assignState) ([]workv1alpha2.TargetCluste
 	return divideReplicasByResource(state.candidates, state.object, policyv1alpha1.ReplicaDivisionPreferenceAggregated)
 }
 
-// assignByStaticWeightStrategy assigns replicas by StaticWeightStrategy.
+/*
+* assignByStaticWeightStrategy assigns a total number of replicas to the selected clusters by the weight list.
+* For example, we want to assign replicas to two clusters named A and B.
+* | Total | Weight(A:B) | Assignment(A:B) |
+* |   9   |   1:2       |     3:6         |
+* |   9   |   1:3       |     2:7         | Approximate assignment
+* Note:
+* 1. If any selected cluster which not present on the weight list will be ignored(different with '0' replica).
+* 2. In case of not enough replica for specific cluster which will get '0' replica.
+ */
 func assignByStaticWeightStrategy(state *assignState) ([]workv1alpha2.TargetCluster, error) {
 	// If ReplicaDivisionPreference is set to "Weighted" and WeightPreference is not set,
 	// scheduler will weight all clusters averagely.
 	if state.strategy.WeightPreference == nil {
 		state.strategy.WeightPreference = getDefaultWeightPreference(state.candidates)
 	}
-	return divideReplicasByStaticWeight(state.candidates, state.strategy.WeightPreference.StaticWeightList, state.object.Replicas)
+	weightList := getStaticWeightInfoList(state.candidates, state.strategy.WeightPreference.StaticWeightList)
```

Comment:
According to your ut, if `state.object.Replicas` is 0, just give a waring log and return it.

Reply:
Acutually, it never happens. Assignment is only revoked when replicas are non-zero.

https://github.com/karmada-io/karmada/blob/master/pkg/scheduler/core/generic_scheduler.go#L181